### PR TITLE
Drools 5456 - Some Drools Projects don't compile with JDK 11

### PR DESCRIPTION
**Pull request by gorky**
https://issues.redhat.com/browse/DROOLS-5456

Updates to build under JDK 11.
This PR makes the following updates to allow the Drools Projects to build under JDK 11: * Updates version.org.eclipse.tycho to 2.3.0 to add support for JDK 11 * Updates the Maven Compile Goal to use the RELEASE nomenclature.

Jenkins run fdb

### CODE REVIEW

**Comment by mareknovotny** [comment]
ok to test

**Comment by mareknovotny** [comment]
jenkins do fdb

**Comment by mariofusco** [comment]
jenkins do fdb

**Review by mariofusco — APPROVED** [review]

**Comment by mariofusco** [comment]
jenkins do fdb

File: pom.xml [hunk 1]
```diff
@@ -484,6 +484,7 @@
     <version.org.jvnet.jaxb2.maven2.maven-jaxb2-plugin>0.14.0</version.org.jvnet.jaxb2.maven2.maven-jaxb2-plugin>
     <version.org.mock-server>5.11.1</version.org.mock-server>
     <version.org.apache.kafka>2.8.1</version.org.apache.kafka>
+    <version.shade.plugin>3.2.4</version.shade.plugin>
```
Comment (tkobayas) [inline pom.xml:487]: @gorky `version.shade.plugin` is defined but it's not referenced in `<pluginManagement>`. Is it a mistake?

Comment (mareknovotny) [reply pom.xml:487]: @tkobayas it is not a mistake. Drools use jboss-parent as parent and plugins are declared there so shade plugin too https://github.com/jboss/jboss-parent-pom/blob/jboss-parent-36/pom.xml#L535-L539

Comment (tkobayas) [reply pom.xml:487]: Oops, sorry. Thank you for pointing that out! @mareknovotny

**Comment by tkobayas** [comment]
FDB : This is being discussed in RHBA Sustaining team chat. So not likely related to this PR.
```
[2022-08-03T21:31:14.120Z] Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.jboss.errai.marshalling.rebind.DefinitionsFactorySingleton
...
[2022-08-03T21:31:14.423Z] [INFO] jBPM Workbench - Showcase .......................... FAILURE [ 55.814 s]
```

**Comment by Ginxo** [comment]
Jenkins run fdb

**Comment by mareknovotny** [comment]
jenkins do fdb

**Pull request merged**